tests/zenko_tests/node_tests/iam_policies/cloudserver/AssumeRole.js

```diff
@@ -0,0 +1,275 @@
+const externalSecretKey1 = 'iTuJdlidzrLipymvAGrLP66Yxghl4NQxLZR3cLlu';
+
+const account2Name = 'iam-policies-assume-role-test-account2';
+const account2Info = {
+    email: `${account2Name}@test.com`,
+};
+const externalAccessKey2 = 'DZMMJUPWIUK8IWXRP0HR';
+const externalSecretKey2 = 'iTuJdlidzrLipymvAGrLP66Yxghl4NQxLZR3cLlv';
+
+const userName = 'iam-policies-assume-role-test-user'
+
+const trustPolicy = JSON.stringify({
+    Version: '2012-10-17',
+    Statement: {
+        Sid: 'test',
+        Effect: 'Allow',
+        Action: 'sts:AssumeRole',
+        Principal: {AWS: '*'},
+    },
+});
+
+const allowAssumeRolePolicy = JSON.stringify({
+    Version: '2012-10-17',
+    Statement: [{
+        Effect: 'Allow',
+        Action: 'sts:AssumeRole',
+        Resource: '*',
+    }],
+})
+let allowAssumeRolePolicyArn = null;
+
+const bucket1 = "bucket1";
+const bucket2 = "bucket2";
+
+let iamAccount1Client = null;
+let iamAccount1Id = null;
+let iamAccount2Client = null;
+let stsClient = null;
+let s3Account1Client = null;
+
+describe('iam policies - cloudserver - AssumeRole - Metadata', () => {
+
+    before(done => {
+        async.series([
+            // create account1, generateAccountAccessKey for it and get account1 iam client
+            next => clientAdmin.createAccount(account1Name, account1Info, (err, res) => {
+                if (err) return next(err);
+                iamAccount1Id = res.account.id;
+                clientAdmin.generateAccountAccessKey(account1Name, () => {
+                        iamAccount1Client = VaultClient.getIamClient(externalAccessKey1, externalSecretKey1);
+                    },
+                    {externalAccessKey: externalAccessKey1, externalSecretKey: externalSecretKey1})
+                next();
+            }),
+            // create account2, generateAccountAccessKey for it, get account2 iam client
+            // create a user under account2, create access key for the user and get user sts client
+            next => clientAdmin.createAccount(account2Name, account2Info, err => {
+                if (err) return next(err);
+                clientAdmin.generateAccountAccessKey(account2Name, () => {
+                        iamAccount2Client = VaultClient.getIamClient(externalAccessKey2, externalSecretKey2);
+                        iamAccount2Client.createUser({UserName: userName}, err => {
+                            if (err) return next(err);
+                            iamAccount2Client.createAccessKey({UserName: userName}, (err, result) => {
+                                if (err) return next(err);
+                                stsClient = getSTSClient(result.AccessKey.AccessKeyId, result.AccessKey.SecretAccessKey);
+                                return next();
+                            })
+                        })
+                    },
+                    {externalAccessKey: externalAccessKey2, externalSecretKey: externalSecretKey2})
+            }),
```

Comment:
Can we flatten these API calls, or avoid using the same variable names in subfunctions?

Reply:
yeah I will flatten them

Reply:
solved: https://github.com/scality/Zenko/pull/1491/commits/b0a74bad0293e3acfc60834bd0773f07410c8f44